rosbag2_examples/rosbag2_examples_cpp/src/simple_bag_player.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Open Source Robotics Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/serialization.hpp"
+#include "rosbag2_cpp/reader.hpp"
+#include "std_msgs/msg/string.hpp"
+
+using namespace std::chrono_literals;
+
+class SimpleBagPlayer : public rclcpp::Node
+{
+public:
+  explicit SimpleBagPlayer(const std::string & bag_filename)
+  : Node("playback_node")
+  {
+    publisher_ = this->create_publisher<std_msgs::msg::String>("chatter", 10);
+
+    timer_ = this->create_wall_timer(
+      100ms,
+      [this]() {return this->timer_callback();}
+    );
+
+    reader_.open(bag_filename);
+  }
+
+private:
+  void timer_callback()
+  {
+    while (reader_.has_next()) {
+      rosbag2_storage::SerializedBagMessageSharedPtr msg = reader_.read_next();
+
+      if (msg->topic_name != "chatter") {
+        continue;
+      }
+
+      rclcpp::SerializedMessage serialized_msg(*msg->serialized_data);
+      std_msgs::msg::String::SharedPtr ros_msg = std::make_shared<std_msgs::msg::String>();
+
+      serialization_.deserialize_message(&serialized_msg, ros_msg.get());
```

Comment:
Do we need to deserialize the message to be published? why can we not do `publish(rclcpp::SerializedMessage)`?

Reply:
One of the requests from #1449 was to show how to serialize/deserialize messages. I also think it would be nice to include (de)serialization in the examples.

How about adding comments to inform that it is also possible to publish serialized messages?

Reply:
I see, as an example, that makes more sense. adding comment would be good to, i can leave this to you. thanks!